support-lib/jni/djinni_support.cpp

```diff
@@ -239,8 +239,13 @@ jint JniEnum::ordinal(JNIEnv * env, jobject obj) const {
 
 LocalRef<jobject> JniEnum::create(JNIEnv * env, jint value) const {
     LocalRef<jobject> values(env, env->CallStaticObjectMethod(m_clazz.get(), m_staticmethValues));
+    jniExceptionCheck(env);
     DJINNI_ASSERT(values, env);
-    return LocalRef<jobject>(env, env->GetObjectArrayElement(static_cast<jobjectArray>(values.get()), value));
+    LocalRef<jobject> result(env,
+                             env->GetObjectArrayElement(static_cast<jobjectArray>(values.get()),
+                                                        value));
+    jniExceptionCheck(env);
+    return std::move(result);
```

Comment:
Yeah, "locate in ROM" works for global/static data.  Not generally for local data, unless you had some kind of a pool of ROM with unique addresses but known values (e.g. a pool of zeroes).  But even that wouldn't apply here since RVO says you have to put it in a location specified by the caller, which can't be ROM since the caller has to specify it without knowing your value yet.